src/include/txn_inline.h

```diff
@@ -338,15 +358,31 @@ __wt_txn_op_delete_commit_apply_timestamps(WT_SESSION_IMPL *session, WT_REF *ref
                 (*updp)->durable_ts = txn->durable_timestamp;
             }
     }
-    page_del = ref->page_del;
-    if (page_del != NULL && page_del->timestamp == WT_TS_NONE) {
-        page_del->timestamp = txn->commit_timestamp;
-        page_del->durable_timestamp = txn->durable_timestamp;
-    }
+
+    __txn_op_delete_commit_apply_page_del_timestamp(session, ref);
 
     WT_REF_UNLOCK(ref, previous_state);
 }
 
+/*
+ * __txn_should_assign_timestamp --
+ *     Some updates may not have any timestamps associated with them, for example, if they were made
```

Comment:
"Some updates may not have any timestamps associated with them" may be change to "We don't apply timestamp to the updates in some cases?"

Reply:
Yeah, that sounds good.